tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/troubleshoot/StrandDumpTest.java

```diff
@@ -145,50 +147,52 @@ private void startProcessAndVerifyStrandDump(ProcessBuilder processBuilder, Map<
+            serverInfoLogReader.removeAllLeechers();
         } catch (InterruptedException | IOException e) {
             throw new BallerinaTestException("Error testing strand dump", e);
         }
     }
 
-    private void verifyObtainedStrandDump(Path expectedOutputFilePath, String obtainedStrandDump)
-            throws BallerinaTestException {
-        List<String> expectedOutputRegEx;
-        try {
-            expectedOutputRegEx = Files.lines(expectedOutputFilePath)
-                    .filter(s -> !s.isBlank()).collect(Collectors.toList());
-        } catch (IOException e) {
-            throw new BallerinaTestException("Failure to read from the expected strand dump output file");
+    private static void populateLeechers(List<LogLeecher> leecherList, Path leecherFilePath,
+                                  ServerLogReader serverInfoLogReader) throws BallerinaTestException {
+        List<String> nonEmptyLines = readFileNonEmptyLines(leecherFilePath);
+        for (String str : nonEmptyLines) {
+            LogLeecher leecher = new LogLeecher(str, true, LogLeecher.LeecherType.INFO);
+            leecherList.add(leecher);
+            serverInfoLogReader.addLeecher(leecher);
         }
+    }
 
-        for (String expectedRegEx : expectedOutputRegEx) {
-            Pattern pattern = Pattern.compile(expectedRegEx);
-            Matcher matcher = pattern.matcher(obtainedStrandDump);
-            if (!matcher.find()) {
-                throw new BallerinaTestException("Failure to obtain the expected strand dump.\nExpected regular " +
-                        "expression: " + expectedRegEx + "\nObtained strand dump:\n" + obtainedStrandDump);
-            }
+    private static void waitForLeechers(List<LogLeecher> logLeechers) throws BallerinaTestException {
+        for (LogLeecher leecher : logLeechers) {
+            leecher.waitForText(TIMEOUT);
         }
     }
 
-    private String getStdOutAsString(Process process) throws BallerinaTestException {
-        String output;
-        InputStream inputStream = process.getInputStream();
-        try (InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
-             BufferedReader buffer = new BufferedReader(inputStreamReader)) {
-            output = buffer.lines().collect(Collectors.joining("\n"));
-        } catch (Exception e) {
-            throw new BallerinaTestException("Error when reading from the stdout ", e);
+    private static List<String> readFileNonEmptyLines(Path filePath) throws BallerinaTestException {
+        try (Stream<String> fileLines = Files.lines(filePath)) {
+            return fileLines.filter(s -> !s.isBlank()).collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new BallerinaTestException("Failure to read from the file: " + filePath);
         }
-        return output;
+    }
+
+    private static boolean isWindowsOS() {
+        return Utils.getOSName().toLowerCase(Locale.ENGLISH).contains("windows");
```

Comment:
We have similar methods used in other classes as well. (Eg. `BMainInstance`, `BServerInstance` etc.) Shall we move this to the `Utils` class and reuse it?